src/Chapter3.hs

```diff
@@ -984,12 +989,8 @@ instance Append [a] where
   append = (++)
 
 instance (Append a) => Append (Maybe a) where
-  append (Just x) Nothing = Just x
-  append Nothing (Just x)  = Just x
-  append (Just x) (Just y) = Just $ append x y
-  append Nothing Nothing = Nothing
-
-
+  append (Just x) a = fmap (append x) a
```

Comment:
This seems incorrect 🤔 
If `a` is nothing, this function will return `Nothing` but we want to return `Just x` here.